internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,456 @@
+
+			pluginManifest := new(manifest)
+			if err := json.Unmarshal(jsonByteArr, &pluginManifest); err != nil {
+				return nil, err
+			}
+
+			return pluginManifest, nil
+		}
+	}
+
+	return nil, errors.Errorf(`failed to find manifest file inside local archive file "%s"`, archivePath)
+}
+
+func getRemoteManifest(owner, name, version string) (*manifest, error) {
+	manifestUrl := fmt.Sprintf("https://api.hub.confluent.io/api/plugins/%s/%s", owner, name)
+	if version != "latest" {
+		manifestUrl = fmt.Sprintf("%s/versions/%s", manifestUrl, version)
+	}
+
+	r, err := http.Get(manifestUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to read manifest file from Confuent Hub")
+	}
+
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	pluginManifest := new(manifest)
+	if err := json.Unmarshal(body, &pluginManifest); err != nil {
+		return nil, err
+	}
+
+	return pluginManifest, nil
+}
+
+func getPluginDirFromFlag(cmd *cobra.Command) (string, error) {
+	if cmd.Flags().Changed("plugin-directory") {
+		pluginDir, err := cmd.Flags().GetString("plugin-directory")
+		if err != nil {
+			return "", err
+		}
+
+		if pluginDir, err = filepath.Abs(pluginDir); err != nil {
+			return "", err
+		}
+
+		if !utils.DoesPathExist(pluginDir) {
+			return "", errors.Errorf(invalidDirectoryErrorMsg, pluginDir)
+		}
+
+		return pluginDir, nil
+	}
+
+	return "", nil
+}
+
+func existingPluginInstallation(pluginDir string, pluginManifest *manifest) ([]string, error) {
+	// Bundled installations
+	if utils.DoesPathExist(filepath.Join(pluginDir, pluginManifest.Name)) {
+		return nil, errors.New("unable to install plugin because it is already bundled")
+	}
+
+	// Other previous installations
+	immediateDirectory := filepath.Join(pluginDir, fmt.Sprintf("%s-%s", pluginManifest.Owner.Username, pluginManifest.Name))
+	uberJar := filepath.Join(pluginDir, fmt.Sprintf("%s-%s.jar", pluginManifest.Name, pluginManifest.Version))
+
+	var installations []string
+	if utils.DoesPathExist(immediateDirectory) {
+		installations = append(installations, immediateDirectory)
+	}
+	if utils.DoesPathExist(uberJar) {
+		installations = append(installations, uberJar)
+	}
+
+	return installations, nil
+}
+
+func uninstall(pathToPlugin string, dryRun, force bool) error {
+	if force {
+		output.Printf("Uninstalling the existing version of the plugin located at \"%s\".\n", pathToPlugin)
+	} else {
+		f := form.New(form.Field{
+			ID:        "confirm",
+			Prompt:    fmt.Sprintf("Do you want to uninstall an existing version of this plugin located at %s?", pathToPlugin),
+			IsYesOrNo: true,
+		})
+		if err := f.Prompt(form.NewPrompt(os.Stdin)); err != nil {
+			return err
+		}
+		if !f.Responses["confirm"].(bool) {
+			return errors.New("previous versions must be uninstalled to continue")
+		}
+	}
+	if dryRun {
+		output.Println("[DRY RUN] Skipping uninstallation of the existing version.")
```

Comment:
How about we always write: "Uninstalled package XXX" but for dry-run mode it's prefixed with `DRY RUN`?